src/cryptonote_core/service_node_list.cpp

```diff
@@ -1520,30 +1587,38 @@ namespace service_nodes
     data.version = get_min_service_node_info_version_for_hf(hf_version);
     {
       std::lock_guard<boost::recursive_mutex> lock(m_sn_mutex);
-      data.quorum_states.reserve(m_quorum_states.size() + m_old_quorum_states.size());
-      for (const auto *qs : {&m_old_quorum_states, &m_quorum_states})
+      for (const quorums_by_height &entry : m_old_quorum_states)
       {
-        for(const auto& kv_pair : *qs)
-        {
-          quorum_for_serialization quorum = {};
-          quorum.version                  = get_min_service_node_info_version_for_hf(hf_version);
-          quorum.height                   = kv_pair.first;
-          quorum_manager const &manager   = kv_pair.second;
+        quorum_for_serialization quorum = serialize_quorum_state(hf_version, entry.height, entry.quorums);
+        data.quorum_states.push_back(std::move(quorum));
+      }
 
-          if (manager.obligations)
-            quorum.quorums[static_cast<uint8_t>(quorum_type::obligations)] = *manager.obligations;
+      if (m_state_history.size() == 1)
+      {
+        data.states.emplace_back(serialize_service_node_state_object(hf_version, m_state_history[0]));
+      }
+      else if (m_state_history.size() >= 2)
+      {
+        for (size_t i = 0; i < (m_state_history.size() - 1); i++)
+        {
+          state_t const &curr = m_state_history[i];
+          state_t const &next = m_state_history[i + 1];
 
-          if (manager.checkpointing)
-            quorum.quorums[static_cast<uint8_t>(quorum_type::checkpointing)] = *manager.checkpointing;
+          if (next.height <= curr.height)
+          {
+            LOG_PRINT_L0("States to serialise are not stored in ascending order by height in DB, failed to store to DB");
+            return false;
```

Comment:
If this happens what happens next?  Does it initiate a rescan?  It seems like a sort of "uh oh, the db is broken" error that is going to be permanent.

Reply:
Fails to store until the next time rescan_starting_from_curr_state occurs so, blockchain reorg or the daemon is restarted at which it'll rescan from the last height that was stored correctly.